.github/workflows/mvn-verify-check.yml

```diff
@@ -30,7 +30,40 @@ env:
     -Dmaven.scaladoc.skip
 
 jobs:
+  cache-dependencies:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ '8']
+    steps:
+      - uses: actions/checkout@v3 # refs/pull/:prNumber/merge
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: ${{ matrix.java }}
+          cache: 'maven'  
+      - run: |
+          set -x
+          max_retry=3; delay=30; i=1
+          while true; do
+            mvn help:evaluate -pl dist -Dexpression=included_buildvers \
+            -DforceStdout -Dmaven.repo.local=$PWD/.m2 -PnoSnapshots -q | tr -d ',' | \
+            xargs -n 1 bash -c 'mvn initialize -pl sql-plugin-api -am -Dbuildver=$1' _ && break || {
+            if [[ $i -le $max_retry ]]; then
+              echo "mvn command failed. Retry $i/$max_retry."; ((i++)); sleep $delay; ((delay=delay*2))
+            else
+              echo "mvn command failed. Exit 1"; exit 1
+            fi
+          }
+          done 
+      - name: Cache local Maven repository
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ env.BRANCH }}-${{ env.DATE }}
```

Comment:
here is an SO for how to set day-precision date https://stackoverflow.com/a/60942437

Reply:
The key is like `Linux-maven-refs/pull/9674/merge-2023-11-29`
Note that we only have 10G space for cache, and our cache is 2.2G. If there are too many branches, the elder ones will be removed.

Reply:
I have to add a step to get current date, since the DATE env var cannot be stored between jobs.

Reply:
> Linux-maven-refs/pull/9674/merge-2023-11-29

The PR branch ref `refs/pull/9674/merge` in the key is not a good idea because it means that all PRs having updates on the same day will not benefit from each other's cache. My suggestion  was the target branch name, e.g. `branch-24.02`, as part of the key. If it is hard to get  let us go back to the sql-plugin-api/pom.xml hash.

Reply:
It's easy to get the target branch info, but I recommend to add **/pom.xml hash to the key, or if there is change in pom, we may miss the change by using key with target branch and date info only.

Reply:
I am OK with that. 

But it is OK to miss some pom changes because
- many pom changes are not even related to dependencies 
-  we generate a cache miss after each midnight which will trigger a new cache rebuild anyhow.
- it is not a correctness issue just a handful dependencies may will be uncached for a day